packages/exec/src/toolrunner.ts

```diff
@@ -392,6 +393,8 @@ export class ToolRunner extends events.EventEmitter {
    * @returns   number
    */
   async exec(): Promise<number> {
+    this.toolPath = await io.which(this.toolPath, true)
```

Comment:
Is this a breaking change?

Reply:
my intention was that it doesnt break anything

for example `which` returns the path if it is already rooted and executable.

however, now that you got me thinking about it more, i think there are some edge cases wrt relative pathing.... i'll add code to handle.

Reply:
ok i added the code above to root unrooted paths that contain relative pathing

in addition to checking the PATH, the which function also handles checking rooted paths

now the changes shouldnt be a breaking change. hard to imagine it would have practically broken anyone, but i'd rather cover edge cases to be on the safe side. Also now the changes align better with the default behavior of `spawn`.

Reply:
i'll add unit tests tomorrow for the unrooted-relative-pathing cases

Reply:
No, it's not a breaking change.  It's additive.  If you were passing in a fully qualified path, which will not changing anything.  It will however add the ability to just pass in a tool name and it just works (as well as fixing some other issues and bugs).

Reply:
@bryanmacfarlane Originally it was breaking, exec of a relative path would no longer work if we automatically `which`. That should be fixed now with @ericsciple's latest change